test/test_sparse_csr.py

```diff
@@ -1319,15 +1323,177 @@ def test_sparse_csr_unary_inplace(self, device, dtype, op):
+
+        # Generate exponent as 0 per the given dtype
+        # Only pow with Scalar exponent is supported as of now, for Sparse CSR
+        if dtype.is_complex:
+            exp = complex(0, 0)
+            with self.assertRaisesRegex(RuntimeError, "not supported"):
+                func(inp, exp, out=out) if is_out else func(inp, exp)
+            return
+        else:
+            exp = 0. if dtype.is_floating_point else 0
+
+            with self.assertRaisesRegex(RuntimeError, err_msg):
+                func(inp, exp, out=out) if is_out else func(inp, exp)
+
+        # Negative exponent is also not supported for Sparse CSR
+        exps = []
+        if dtype.is_floating_point:
+            exps = [-0.0, -2.5]
+        elif dtype in [torch.uint8, torch.int8, torch.int16, torch.int32, torch.int64]:
+            exps = [-0, -1]
+
+        for exp in exps:
+            with self.assertRaisesRegex(RuntimeError, err_msg):
+                func(inp, exp, out=out) if is_out else func(inp, exp)
+
+        # Also check for exponent as boolean (False)
+        with self.assertRaisesRegex(RuntimeError, "not supported"):
+            func(inp, False)
+
+    @ops(sparse_csr_pow)
+    def test_pow_scalar(self, device, dtype, op):
+        # Make sure that the op fails for exponent as 0 and for complex inputs
+        self._test_pow_check_error(op, device, dtype, is_out=False)
+        # _test_pow_check_error already checks for an error if dtype is complex
+        # hence early return since complex isn't supported for Sparse CSR (pow)
+        if dtype.is_complex:
+            return
+
+        samples = op.sample_inputs(device, dtype)
+
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        for sample in samples:
+            assert torch.is_tensor(sample.input)
+            if sample.input.is_complex:
+                continue
+
+            # Sparse CSR only supports 2D tensors as inputs
+            if sample.input.ndim != 2:
+                continue
+
+            # pow(Tensor, Tensor) is not supported for Sparse CSR yet.
+            if torch.is_tensor(sample.args[0]):
+                continue
+
+            expected = op(sample.input, *sample.args, **sample.kwargs)
+            assert torch.is_tensor(expected)
+            output = op(sample.input.to_sparse_csr(), *sample.args, **sample.kwargs)
+            assert torch.is_tensor(output)
+
+            self.assertEqual(output.to_dense(), expected)
+
+    @ops(sparse_csr_pow)
+    def test_pow_scalar_out(self, device, dtype, op):
+        if not op.supports_out:
+            self.skipTest("Skipped! Out not supported")
+
+        samples = op.sample_inputs(device, dtype)
+
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        # Make sure that the op fails for exponent as 0 and for complex inputs
+        self._test_pow_check_error(op, device, dtype, is_out=True)
```

Comment:
This is repeated in `test_pow_scalar`, `test_pow_scalar_out`, and `test_pow_scalar_inplace` with same arguments. It's better to make `_test_pow_check_error` a separate single test.